canonical/maas-ui
# feat: add MAAS new layout side panel

## Done

- move all header forms into maas new layout side panel

### Note
⚠️ [rename of headerContent, setHeaderContent to sidePanelContent, setSidePanelContent](https://warthogs.atlassian.net/browse/MAASENG-1246) will be done in a separate PR to make this easier to review

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Verify that all forms previously placed in a header are displayed correctly in a sidebar
   - devices -> add device
   - controllers -> take action forms
   - controller details - take action
   - machines -> take action forms
   - machine details -> take action forms
   - KVM -> LXD, KVM - Virsh
- Verify that you can step through Add LXD form and all steps are displayed correctly

## Fixes

Fixes: [MAASENG-1229](https://warthogs.atlassian.net/browse/MAASENG-1229)

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

<img width="1673" alt="image" src="https://user-images.githubusercontent.com/7452681/212726731-fb865179-db2c-4fc6-b70f-3202e6ca697f.png">
<img width="1577" alt="image" src="https://user-images.githubusercontent.com/7452681/212726055-78d2102b-8b45-404e-b503-557d1d2acbab.png">


[MAASENG-1229]: https://warthogs.atlassian.net/browse/MAASENG-1229?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
